app/routes/login-callback/route.tsx

```diff
@@ -0,0 +1,37 @@
+import { LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { commitSession, getSession, initializeEcomApiForRequest } from '~/lib/ecom/session';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+    const session = await getSession(request.headers.get('Cookie'));
+    const storedOauthData = session.get('oAuthData');
+
+    if (storedOauthData === undefined) {
+        return redirect('/login');
+    }
+
+    const api = await initializeEcomApiForRequest(request);
+    const wixClient = api.getWixClient();
+
+    const returnedOAuthData = wixClient.auth.parseFromUrl(request.url, 'query');
+    if (returnedOAuthData.error) {
+        throw new Error(`Error: ${returnedOAuthData.errorDescription}`);
+    }
+
+    const memberTokens = await wixClient.auth.getMemberTokens(
+        returnedOAuthData.code,
+        returnedOAuthData.state,
+        storedOauthData,
+    );
+    wixClient.auth.setTokens(memberTokens);
+
+    session.set('wixEcomTokens', memberTokens);
+
+    const redirectUrl = new URL(storedOauthData.originalUri || new URL(request.url).host);
```

Comment:
we are creating a new instance of `URL` class here. if value is empty string - it will fail with errror, so I'd like to have `||` in this case
![image](https://github.com/user-attachments/assets/79e3e600-404c-4dca-9039-f1e78a6ad4c4)